### DISCUSSION TITLE
Image FitHeight

### DISCUSSION THREAD

**Discussion by HightowerCZ** [QuestPDF/QuestPDF#830] [Q&A] [answered]
Hi,

I am trying to have a row container with two items. The first item is a Column with several items. The second item in the row is an image which should Fit its height to have the same height as the first section (with columns container). If I use Image.FitUnproportionally(), it have a correct height, but if I use FitHeight, it extends height of the first item.

FitUnproportionally()
![image](https://github.com/QuestPDF/QuestPDF/assets/22144797/616c26bb-64c1-4b12-94a2-46421e3f61d9)

FitHeight()
![image](https://github.com/QuestPDF/QuestPDF/assets/22144797/872ceab1-5fc2-46c3-a1f1-b7900d3ca7f1)

How can I set the image the same height as the first section? I don't know the size of the first section since the user can change fonts?

This is the code I use:

```
container.ShrinkVertical().Row(r =>
{
   r.AutoItem().Column(sectionColumn =>
  {
        sectionColumn.Item().Row(r =>
        {
            r.AutoItem().Width(keyWidth, Unit.Millimetre).AlignRight().Text(Key).Bold();
            var valueItem = r.AutoItem().AlignLeft().Text(Value);
        });
  });

  r.RelativeItem().Image(model.SignatureImage).FitHeight(); //.FitUnproportionally();
}
```

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

The issue is that `FitHeight()` fills the full available height and is meant to be paired with a height constraint, so without one it expands against your `ShrinkVertical` and grows the row. Use `FitArea()` instead — it scales the image proportionally to fit within the space your `ShrinkVertical().Row(...)` already bounds to the text column's height, without overgrowing. See the [Image](https://www.questpdf.com/api-reference/image/basics.html) documentation for all fitting options. For an exact match in more dynamic layouts, you can also constrain the image with a `MaxHeight`.

Should you encounter any further problems, please don't hesitate to open a new issue.
Thank you!
